release_util/management/commands/__init__.py

```diff
@@ -0,0 +1,203 @@
+            call_command("migrate",
+                         interactive=False,
+                         stdout=out,
+                         database=self.database_name)
+        except (CommandError, DatabaseError) as e:
+            time_to_fail = self.timer() - start
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+
+            # Parse the output to discover the last migration being applied before the exception.
+            # Mark migrations:
+            # - before exception migration as success
+            # - exception migration as failed
+            # - any remaining exceptions as unapplied
+            # Use the initialized list of migrations to determine the ordering.
+            for line in out.getvalue().split('\n'):
+                line = _remove_escape_characters(line).strip()
+                if line.startswith('Applying'):
+                    if line.endswith('OK'):
+                        # A migration succeeded.
+                        app, migration = self.to_apply.pop(0)
+                        self.migration_state['success'].append({
+                            'migration': [app, migration],
+                            'output': line,
+                        })
+                    else:
+                        # This migration failed.
+                        failed_app, failed_migration = self.to_apply.pop(0)
+                        self.migration_state['failure'] = {
+                            'migration': [failed_app, failed_migration],
+                            'duration': time_to_fail,
+                            'output': out.getvalue(),
+                            'traceback': repr(traceback.format_exception(exc_type, exc_value, exc_traceback)),
+                        }
+                        break
+            # Any leftover migrations were *not* applied.
+            while self.more_to_apply():
+                app_migration = self.to_apply.pop(0)
+                self.migration_state['unapplied'].append(app_migration)
+
+            # Find the apps that have been applied -or- failed.
+            # Include their initial migrations as commands.
+            self._add_rollback_commands()
+            raise CommandError("Migration failed for app '{}' - migration '{}'.\n".format(failed_app, failed_migration))
+
+        # All migrations succeeded.
+        time_to_apply = self.timer() - start
+        while self.to_apply:
+            app, migration = self.to_apply.pop(0)
+            self.migration_state['success'].append({
+                'migration': [app, migration],
+                'duration': time_to_apply,
+                'output': out.getvalue(),
+            })
+
+    def apply_all_one_by_one(self):
+        """
+        DEPRECATED:
+        Apply all the migrations, executing each migration individually.
+        """
+        while self.more_to_apply():
+            self._apply_next()
+        self._add_rollback_commands()
+
+    def apply_all_together(self):
```

Comment:
That's fine - I'll change the name.

I'm leaving the old method out of caution - so that I can simply point to the other mgmt command if something is wrong with the new one. After the new mgmt command is in use for long enough, I'll go back and remove the deprecated code.